### ISSUE TITLE
Syntax error bug in setup.ion, etc.

### ISSUE THREAD

**Issue by covercash2** [redox-os/ion#607]
i was having trouble getting setup.ion to run, and i found what appears to be a bug.

`if not exists $(rustc --version | grep nightly)
    echo hello
end`

outputs:
`ion: syntax error: '(' at position 29 is out of place`

and it doesn't execute the inner block. i haven't been able to reproduce the error with other commands inside the `$()` statement.

it also has hard to define effects on the setup.ion script. mostly it just doesn't work. removing the rustc version check seems to solve a lot of it. the other problems may merit another issue, but i'm still working on notes for those.

### version stuff

i've tested this on Arch, a Debian 9 VM, and macOS with termite, iTerm2, and Terminal.app.

`rustc 1.23.0-nightly (827cb0d61 2017-11-26)`

`ion 1.0.5 (x86_64-apple-darwin)`
`rev bfe7dffe507bb2565cf9feb92b46da6059b7757b`

**Comment by mmstick** [comment]
I will look into this now.

**Comment by mmstick** [comment]
For some reason, the output of the process expansion is being provided to a StatementSplitter, hence the error. This should not be happening. Will investigate further.

**Comment by mmstick** [comment]
I have found the issue, and I am now working on a fix.

**Comment by mmstick** [comment]
This issue is now resolved. It was caused due to the not/and/or builtins taking their arguments, joining them as a string, and then sending them back to the front of the line to the **on_command()** method to be reparsed by the statement splitter and re-argument-splitted. So I've implemented the logic to just take the argument array and convert it into a pipeline manually and executing that pipeline as is. Solves the issue.
